### DISCUSSION TITLE
[firestore-bigquery-export] - How to specify Firestore database

### DISCUSSION THREAD

**Discussion by gmyrianthous** [firebase/extensions#2200] [Q&A]
I'm trying to enable Firestore to BigQuery replication, by using the [Stream Firestore to BigQuery][1] Firebase extension that will be in charge of the continuous replication process. On top of it, and since the extension currently does not allow for backfilling options (due to a bug that resulted in the removal of this feature from latest releases), I would like to run [@firebaseextensions/fs-bq-import-collection][2] script to perform the backfilling and replicating existing data from Firestore to BigQuery. 

Things work fine, as far as I replicate data from a collection that resides in the `(default)` database. However, when I attempt to perform the replication (with either of the tools) from a collection in a database other than `(default)`, no records are being replicated. 

I believe, I'm probably doing something wrong when it comes down to specifying the collection path. None of these tools provide the option to specify the specific database. How do I specify a collection path, that also includes the database details ?

What I've tried so far, assuming that:
 - project ID: `my-project`
 - Firestore db name: `my-db`
 - Firestore collection name: `my-coll`

1. Specifying just the collection name

```
my-coll 
 
``` 
this would work if `my-coll` was in `(default)` db. 

2. Specifying both db and collection
```
my-db/my-coll 
```
this gives me an error from `@firebaseextensions/fs-bq-import-collection`:
```
The collection path does not match the regular expression provided
```
3. Specifying project, db, collection
```
projects/my-project/databases/my-db/documents/my-coll
```
but once again, I get the error
```
The collection path does not match the regular expression provided
```

Anyone who can shed some light here? It makes no sense not to be able to specify a db at all. What if I had collections with the same name residing in different dbs? 

Is there a good reason why it was decided to [hardcode (!!!!) the database name](https://github.com/firebase/extensions/blob/firestore-bigquery-export-v0.1.55/firestore-bigquery-export/functions/src/config.ts#L37) instead of including it in the prompt? 

  [1]: https://extensions.dev/extensions/firebase/firestore-bigquery-export
  [2]: https://github.com/firebase/extensions/blob/next/firestore-bigquery-export/README.md

**Comment by IM-CrazyDeveloper** [comment]
As per this discussion  https://github.com/firebase/extensions/pull/2122 DB name is not supported with current version of extension.

**Comment by shravya-ujire** [comment]
do we have any solution here?
